### PR TITLE
CCD-3535 CCD-3562 extra CR&D FTAs 

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [CCD-3587](https://tools.hmcts.net/jira/browse/CCD-3587) _"Add data-store FTAs to test callback behaviour for CCD-3535 & CCD-3562"_
  * [CCD-3535](https://tools.hmcts.net/jira/browse/CCD-3535) _"Adjust callback checks that verify TTL content has not been changed"_
  * [CCD-3562](https://tools.hmcts.net/jira/browse/CCD-3562) _"Adjust callback TTL checks to permit when TTL is missing from response"_

### Change description ###

Functional Tests for recent Case Retain & Dispose tickets

> *Note*: this PR is dependent on: 
> * hmcts/ccd-test-definitions#50
> * hmcts/ccd-test-stubs-service#166
and 
> * hmcts/ccd-test-definitions#51

> *Note*: this PR was tested in def-file isolation using: 
> * hmcts/ccd-definition-store-api#1264

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -19,7 +19,8 @@ def branchesToSync = ['demo', 'ithc', 'perftest']
 
 // Variables to switch pipeline logic and wiring per type of build
 
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
+// TODO: remove TEMPORARY OVERRIDE FOR CR&D, reinstate:     def definitionStoreDevelopPr = "PR-1264"
+def definitionStoreDevelopPr = "PR-1264"  // This doesn't change frequently, but when it does, only change this value.
```
Comment (mattnayler) [inline Jenkinsfile_CNP:23]: These values should be restored prior to merger.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -103,7 +104,8 @@ env.DEFAULT_COLLECTION_ASSERTION_MODE="UNORDERED"
 env.MAX_NUM_PARALLEL_THREADS=6
 env.BEFTA_S2S_CLIENT_ID = "ccd_gw"
 env.ROLE_ASSIGNMENT_API_GATEWAY_S2S_CLIENT_ID = "ccd_data"
-env.BEFTA_TEST_STUB_SERVICE_BASE_URL = "http://ccd-test-stubs-service-aat.service.core-compute-aat.internal"
+// TODO: remove TEMPORARY OVERRIDE FOR CR&D, reinstate:     env.BEFTA_TEST_STUB_SERVICE_BASE_URL = "http://ccd-test-stubs-service-aat.service.core-compute-aat.internal"
+env.BEFTA_TEST_STUB_SERVICE_BASE_URL = "http://ccd-test-stubs-service-pr-166.service.core-compute-preview.internal"
```
Comment (mattnayler) [inline Jenkinsfile_CNP:108]: These values should be restored prior to merger.

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 3]
```diff
@@ -27,7 +27,8 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.auth.password}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    # TODO: remove TEMPORARY OVERRIDE FOR CR&D, reinstate:     DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1264.service.core-compute-preview.internal/
```
Comment (mattnayler) [inline charts/ccd-data-store-api/values.preview.template.yaml:31]: These values should be restored prior to merger.

File: build.gradle [hunk 4]
```diff
@@ -277,7 +277,7 @@ dependencies {
     compile 'org.jooq:jool-java-8:0.9.14'
     compile 'com.github.hmcts:ccd-case-document-am-client:1.7.1'
 
-    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.1'
+    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.2_CCD-3535'
```
Comment (mattnayler) [inline build.gradle:280]: This needs to be updated to a full release version of the hmcts/ccd-test-defintions before going into master. see https://github.com/hmcts/ccd-test-definitions/pull/50.

**Review by sullivangareth — APPROVED** [review]

File: build.gradle (hunk 4 above)
Comment (mattnayler) [reply build.gradle]: Resolved in 6969fc3f64282b8a73ccbd9991759ca861962bb7

Comment (marianadpereira) [reply build.gradle]: Done in [6969fc3](https://github.com/hmcts/ccd-data-store-api/pull/2058/commits/6969fc3f64282b8a73ccbd9991759ca861962bb7)

File: src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature [hunk 5]
```diff
@@ -395,11 +486,497 @@ Feature: F-1015: Update Case - Start Case Event - Update Code for TTL
...
+      And it is submitted to call the [Start event creation process to update a case (v1_ext caseworker)] operation of [CCD Data Store]
+
+     Then a positive response is received
+      And the response has all other details as expected
+      And the response [contains the TTL.SystemTTL for the case, that has been set to 20 days from today]
+      And the response [does not contain the TTL.OverrideTTL as removed by callback (null -> missing)]
+      And the response [contains the adjusted TTL.Suspended from the callback (Yes -> YES)]
+
+
+  @S-1015.35 #CCD-3572
```
Comment (marianadpereira) [inline src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature:568]: Tag should be #3562

File: Jenkinsfile_CNP (hunk 2 above)
Comment (marianadpereira) [reply Jenkinsfile_CNP]: Fixed in [620a75a](https://github.com/hmcts/ccd-data-store-api/pull/2058/commits/620a75a1ae2c193d4cb5bea24a73c94f8643bf03)

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [inline Jenkinsfile_CNP:23]: These values should be restored prior to merger.

Comment (mattnayler) [reply Jenkinsfile_CNP:23]: https://github.com/hmcts/ccd-data-store-api/pull/2058#discussion_r932991036

File: src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature (hunk 5 above)
Comment (mattnayler) [reply src/aat/resources/features/F-1015 - Start Case Event TTL/F-1015.feature]: Resolved in d89cd500bb25a3455e03fcbc3f01841bb17179af

File: build.gradle [hunk 6]
```diff
@@ -277,7 +277,7 @@ dependencies {
     compile 'org.jooq:jool-java-8:0.9.14'
     compile 'com.github.hmcts:ccd-case-document-am-client:1.7.1'
 
-    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.1'
+    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.19.3-prerelease'
```
Comment (mattnayler) [inline build.gradle:280]: This needs to be updated to a full release version of the hmcts/ccd-test-defintions before going into master. see hmcts/ccd-test-definitions#51.

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [reply Jenkinsfile_CNP]: Resolved in f31e8b632686248b0e1f8844f8fd8c34a4719ac7.

**Review by marianadpereira — APPROVED** [review]

File: build.gradle (hunk 6 above)
Comment (mattnayler) [reply build.gradle]: Resolved in f31e8b632686248b0e1f8844f8fd8c34a4719ac7.

Comment (marianadpereira) [reply build.gradle]: Done in [f31e8b6](https://github.com/hmcts/ccd-data-store-api/pull/2058/commits/f31e8b632686248b0e1f8844f8fd8c34a4719ac7)

File: charts/ccd-data-store-api/values.preview.template.yaml (hunk 3 above)
Comment (marianadpereira) [reply charts/ccd-data-store-api/values.preview.template.yaml]: Done in [95b83a4](https://github.com/hmcts/ccd-data-store-api/pull/2058/commits/95b83a40aac320394117e9162b5b69b3243f1502)

**Pull request merged**